frontend/src/components/common/ConfirmDialog.tsx

```diff
@@ -1,28 +1,41 @@
+import { Checkbox } from '@mui/material';
+import Box from '@mui/material/Box';
 import Button from '@mui/material/Button';
 import MuiDialog, { DialogProps as MuiDialogProps } from '@mui/material/Dialog';
 import DialogActions from '@mui/material/DialogActions';
 import DialogContent from '@mui/material/DialogContent';
 import DialogContentText from '@mui/material/DialogContentText';
-import React, { ReactNode } from 'react';
+import React from 'react';
 import { useTranslation } from 'react-i18next';
 import { DialogTitle } from './Dialog';
 
 export interface ConfirmDialogProps extends MuiDialogProps {
   title: string;
-  description: ReactNode;
+  description: string | React.ReactNode;
+  checkboxDescription?: string;
```

Comment:
Please document this field?

(If you can document the other ones whilst you're here... that would be nice to have).

Reply:
@vyncent-t I know we talked about adding this prop but in the end I don't think this is too specific a behavior to have in the component. Maybe we can pass a fragment that includes the description and the checkbox as part of the description. I will comment below why we don't need to make Agree insensitive depending on the check mark.